databricks-sdk-java/src/main/java/com/databricks/sdk/core/error/AbstractErrorMapper.java

```diff
@@ -0,0 +1,42 @@
+package com.databricks.sdk.core.error;
+
+import com.databricks.sdk.core.DatabricksError;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+abstract class AbstractErrorMapper {
+  @FunctionalInterface
+  protected interface ErrorCodeRule {
+    DatabricksError create(String message, List<ErrorDetail> details);
+  }
+
+  @FunctionalInterface
+  protected interface StatusCodeRule {
+    DatabricksError create(String errorCode, String message, List<ErrorDetail> details);
+  }
+
+  public DatabricksError apply(int code, ApiErrorBody errorBody) {
+    String message = errorBody.getMessage();
+    String errorCode = errorBody.getErrorCode();
+    List<ErrorDetail> details = errorBody.getErrorDetails();
+    if (errorCodeMapping.containsKey(errorCode)) {
+      return errorCodeMapping.get(errorCode).create(message, details);
+    }
+    if (statusCodeMapping.containsKey(code)) {
+      return statusCodeMapping.get(code).create(errorCode, message, details);
+    }
```

Comment:
Error codes always take precedence over status codes. The same status code could be used to represent different error codes. Because of this, we don't need to store status code there.